presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestAnalyzer.java

```diff
@@ -217,6 +217,49 @@ public void testGroupByInvalidOrdinal()
     }
 
     @Test
+    public void testGroupByWithSubquerySelectExpression()
+            throws Exception
+    {
+        analyze("SELECT (SELECT t1.a) FROM t1 GROUP BY a");
+        analyze("SELECT (SELECT a) FROM t1 GROUP BY t1.a");
```

Comment:
add test (should pass):
```
SELECT a as b FROM t1 GROUP BY t1.a ORDER BY (SELECT b)
```

and negative test (since we don't support the feature yet)
```
SELECT (SELECT a*2) FROM t1 GROUP BY a*2
```

another negative test
```
SELECT a*2 FROM t1 GROUP BY t1.a*2
```